src/App.vue

```diff
@@ -25,7 +29,11 @@ export default defineComponent({
   },
   props: {
     productPrice: Number,
-    lang: String,
+    lang: String as () => LanguageCodeEnum,
+    theme: {
+      type: String as () => ThemeEnum,
+      default: (props: AppProps) => props.lang,
```

Comment:
When testing different themes in development this default prop can be changed. e.g: 
```
default: (props: AppProps) => 'qmc',
```
or
```
default: (props: AppProps) => 'farmers',
```